x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -243,6 +244,23 @@ private static String[] enterpriseSearchAcknowledgementMessages(OperationMode cu
         return Strings.EMPTY_ARRAY;
     }
 
+    private static String[] esqlAcknowledgementMessages(OperationMode currentMode, OperationMode newMode) {
+        switch (newMode) {
+            case BASIC:
+            case STANDARD:
+            case GOLD:
+                switch (currentMode) {
+                    case PLATINUM:
```

Comment:
Not sure I understand the logic here. Is it the same as:
```
if((currentMode == ENTERPRISE || currentMode == TRIAL) && newMode != ENTERPRISE && newMode != TRIAL) {
                        return new String[] { "ES|QL cross-cluster search requires an enterprise license." };
}
```
? Maybe some comment spelling it out would be helpful - the intent of the nested switch is not clear to me. 

Also not sure what's the difference between returning EMPTY_ARRAY and `return new String[] {};` - isn't it the same?

Reply:
> Also not sure what's the difference between returning EMPTY_ARRAY ...

Probably a refactoring mistake?

Reply:
And added tests for it.